src/ebonite/ext/lightgbm/model.py

```diff
@@ -6,30 +6,36 @@
 import lightgbm as lgb
 
 from ebonite.core.analyzer.base import TypeHookMixin
-from ebonite.core.analyzer.model import ModelHook
-from ebonite.core.objects import ModelWrapper
+from ebonite.core.analyzer.model import BindingModelHook
+from ebonite.core.objects import ModelIO, ModelWrapper
 from ebonite.core.objects.artifacts import Blobs, LocalFileBlob
 from ebonite.core.objects.wrapper import FilesContextManager
 
 
-class LightGBMModelWrapper(ModelWrapper):
+class LightGBMModelIO(ModelIO):
     """
-    :class:`.ModelWrapper` implementation for `lightgbm.Booster` type
+    :class:`.ModelIO` implementation for `lightgbm.Booster` type
     """
     model_path = 'model.lgb'
 
     @contextlib.contextmanager
-    @ModelWrapper.with_model
-    def _dump(self) -> FilesContextManager:
-        model: lgb.Booster = self.model
+    def dump(self, model: lgb.Booster) -> FilesContextManager:
         with tempfile.TemporaryDirectory(prefix='ebonite_lightgbm_dump') as f:
             path = os.path.join(f, self.model_path)
             model.save_model(path)
             yield Blobs({self.model_path: LocalFileBlob(path)})
 
-    def _load(self, path):
+    def load(self, path):
         model_file = os.path.join(path, self.model_path)
-        self.model = lgb.Booster(model_file=model_file)
+        return lgb.Booster(model_file=model_file)
+
+
+class LightGBMModelWrapper(ModelWrapper):
+    """
+    :class:`.ModelWrapper` implementation for `lightgbm.Booster` type
+    """
+    def __init__(self):
+        super().__init__(LightGBMModelIO())
```

Comment:
just an idea, maybe abc like this will make things easier
```
class ModelWrapperWithIO(ModelWrapper):
    io_class = None
    def __init__(self):
        super().__init__(self.io_class())

class LightGBMModelWrapper(ModelWrapperWithIO):
    io_class = LightGBMModelIO
```

Reply:
This introduces additional class which should be documented and exported in `__init__` for extensions while providing too little value. So I propose to leave as it is.